File: app/javascript/channels/site_channel.js
import consumer from "channels/consumer"

const siteId = window.location.pathname.match(/\/sites\/(\d+)/)?.[1];
console.log(siteId)

consumer.subscriptions.create("SiteChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
  }
});

File: app/channels/site_channel.rb
class SiteChannel < ApplicationCable::Channel
  def subscribed
    @siteName = params[:siteName]
    @stream = "sites.#{@siteName}"

    stream_from @stream
  end

  def receive(data)
  end

  def unsubscribed
    # Cleanup if needed when unsubscribed
  end
end

